[Docs] add ngramXYZ functions

Blargian opened the pull request:
Closes [#1986](https://github.com/ClickHouse/clickhouse-docs/issues/1986) as part of the project to document missing functions ([#1833](https://github.com/ClickHouse/clickhouse-docs/issues/1833)) or to update functions with incomplete documentation.

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes

- [X] Documentation is written (mandatory for new features)
- [X] ngramDistanceCaseInsensitive
- [X] ngramDistanceCaseInsensitiveUTF8
- [X] ngramDistanceUTF8
- [x] ngramSearchCaseInsensitive
- [x] ngramSearchCaseInsensitiveUTF8
- [x] ngramSearchUTF8

Blargian committed:
Update ngramDistance functions

Blargian committed:
Update ngramSearch functions

robot-ch-test-poll3 commented:
<!-- automatic status comment for PR #62111 from Blargian/ClickHouse:document_ngramXYZ -->
*This is an automated comment for commit 380788ea30ff2aba639afc5f3568bbe599d4d835 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62111/380788ea30ff2aba639afc5f3568bbe599d4d835/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

Blargian committed:
Fix spelling mistake

insensitve -> insensitive

Blargian committed:
Add ngram variants to aspell-dict.txt

added the following variants to aspell-dict.txt :

ngramDistanceCaseInsensitive
ngramDistanceCaseInsensitiveUTF
ngramDistanceUTF
ngramSearchCaseInsensitive
ngramSearchCaseInsensitiveUTF
ngramSearchUTF

Blargian committed:
Add examples for UTF8 variants of ngramDistance

Blargian committed:
Correct link to string literal

Blargian committed:
Small fixes

Blargian committed:
Merge branch 'master' into document_ngramXYZ

serxa reviewed (approved):
Thank you. LGTM
